src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -140,6 +140,13 @@ public override
                     if (attrs.Length > 0)
                         return attrs[0].Value;
                 }
+                else
+                {
+                    /* look for a default value encoded using a custom attribute */
+                    CustomConstantAttribute[] attrs = (CustomConstantAttribute[])GetCustomAttributes(typeof(CustomConstantAttribute), false);
```

Comment:
Is the logic above still needed?
What should happen if there is more than 1 instance of CustomConstantAttribute attached to the parameter?

Reply:
This is my mistake, I have overlooked the scenario of attaching more than one `CustomConstantAttribute` to the parameter. 
The coreCLR implementation seems to handle this ambiguously. For example, in case of `DateTime` parameter type, it always favours the `DateTimeConstant` over any other `CustomConstant` attributes attached. Which is different from handling the case for `decimal` parameter type, where it picks the value specified by the first `CustomConstant` attribute attached instead (more info here: https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs#L311-#L317)

Since such scenario is not covered in the test cases, I will investigate this further and include additional test cases to align behaviours between coreCLR and mono.

Reply:
@jkotas how does nativeaot handle such cases?

Reply:
Both CoreCLR and nativeaot pick one at random.